### PR TITLE
Extra functionalities to WebhookUtil

### PR DESCRIPTION
**Pull request by LOOHP**
This PR adds methods in WebhookUtil to:
- Send multiple embeds
- Add attachments
- Add interactions
- Edit existing webhook messages
- Execute webhook on the same thread instead of async

### CODE REVIEW

**Comment by LOOHP** [comment]
oops

File: src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java [hunk 1]
```diff
@@ -143,33 +179,124 @@ public static void deliverMessage(TextChannel channel, OfflinePlayer player, Str
...
+
+    public static void editMessage(TextChannel channel, String editMessageId, String message, Collection<? extends MessageEmbed> embeds, Map<String, InputStream> attachments, Collection<? extends ActionRow> interactions) {
+        executeWebhook(channel, null, null, editMessageId, message, embeds, attachments, interactions, true, true);
+    }
+
+    public static void editMessage(TextChannel channel, String editMessageId, String message, Collection<? extends MessageEmbed> embeds, Map<String, InputStream> attachments, Collection<? extends ActionRow> interactions, boolean scheduleAsync) {
+        executeWebhook(channel, null, null, editMessageId, message, embeds, attachments, interactions, true, scheduleAsync);
+    }
+
+    private static void executeWebhook(TextChannel channel, String webhookName, String webhookAvatarUrl, String editMessageId, String message, Collection<? extends MessageEmbed> embeds, Map<String, InputStream> attachments, Collection<? extends ActionRow> interactions, boolean allowSecondAttempt, boolean scheduleAsync) {
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java:250]: The attachments InputStreams aren't closed if there's a return; or the code execution failes

File: src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java [hunk 2]
```diff
@@ -143,33 +179,124 @@ public static void deliverMessage(TextChannel channel, OfflinePlayer player, Str
...
         if (channel == null) return;
 
-        String webhookUrl = getWebhookUrlToUseForChannel(channel);
-        if (webhookUrl == null) return;
+        String webhookUrlForChannel = getWebhookUrlToUseForChannel(channel);
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java:253]: Not sure if it matters but this is blocking and isn't inside the part that is effected by `scheduleAsync`

**Review by Vankka — CHANGES_REQUESTED** [review]
Couple of inline comments, otherwise looks good

File: src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java (hunk 2 above)
Comment (LOOHP) [reply src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java:253]: I just kept `async=true` the same as how it used to work.

Comment (LOOHP) [reply src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java:263]: Is it ok to leave it like this?

Comment (Vankka) [reply src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java:263]: sure

File: src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java [hunk 3]
```diff
@@ -143,33 +180,142 @@ public static void deliverMessage(TextChannel channel, OfflinePlayer player, Str
...
+        if (webhookUrlForChannel == null) {
+            if (attachments != null) {
+                attachments.values().forEach(inputStream -> {
+                    try {
+                        inputStream.close();
+                    } catch (IOException ignore) {
+                    }
+                });
+            }
+        }
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/WebhookUtil.java:273]: You accidentally removed the `return;`s

**Review by Vankka — APPROVED** [review]

**Pull request merged**